sdk/include/opentelemetry/sdk/trace/span_limits.h

```diff
@@ -0,0 +1,119 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+#include "opentelemetry/common/spin_lock_mutex.h"
+#include "opentelemetry/nostd/shared_ptr.h"
+#include "opentelemetry/version.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace trace
+{
+/**
+ * Collection of span limits configurations.
+ */
+class SpanLimits
+{
+public:
+  /**
+   * Setter methods of configuration params
+   */
+  static void SetAttributeCountLimit(int attributeCountLimit)
+  {
+    std::lock_guard<opentelemetry::common::SpinLockMutex> guard(GetLock());
+    GetSpanLimits()->AttributeCountLimit = attributeCountLimit;
+  }
+
+  static void SetEventCountLimit(int eventCountLimit)
+  {
+    std::lock_guard<opentelemetry::common::SpinLockMutex> guard(GetLock());
+    GetSpanLimits()->EventCountLimit = eventCountLimit;
+  }
+
+  static void SetLinkCountLimit(int linkCountLimit)
+  {
+    std::lock_guard<opentelemetry::common::SpinLockMutex> guard(GetLock());
+    GetSpanLimits()->LinkCountLimit = linkCountLimit;
+  }
+
+  static void SetAttributePerEventCountLimit(int attributePerEventCountLimit)
+  {
+    std::lock_guard<opentelemetry::common::SpinLockMutex> guard(GetLock());
+    GetSpanLimits()->AttributePerEventCountLimit = attributePerEventCountLimit;
+  }
+
+  static void SetAttributePerLinkCountLimit(int attributePerLinkCountLimit)
+  {
+    std::lock_guard<opentelemetry::common::SpinLockMutex> guard(GetLock());
+    GetSpanLimits()->AttributePerLinkCountLimit = attributePerLinkCountLimit;
+  }
+
+  /**
+   * Getter methods of configuration params
+   */
+  static int GetAttributeCountLimit()
+  {
+    std::lock_guard<opentelemetry::common::SpinLockMutex> guard(GetLock());
+    return GetSpanLimits()->AttributeCountLimit;
+  }
+
+  static int GetEventCountLimit()
+  {
+    std::lock_guard<opentelemetry::common::SpinLockMutex> guard(GetLock());
+    return GetSpanLimits()->EventCountLimit;
+  }
+
+  static int GetLinkCountLimit()
+  {
+    std::lock_guard<opentelemetry::common::SpinLockMutex> guard(GetLock());
+    return GetSpanLimits()->LinkCountLimit;
+  }
+
+  static int GetAttributePerEventCountLimit()
+  {
+    std::lock_guard<opentelemetry::common::SpinLockMutex> guard(GetLock());
+    return GetSpanLimits()->AttributePerEventCountLimit;
+  }
+
+  static int GetAttributePerLinkCountLimit()
+  {
+    std::lock_guard<opentelemetry::common::SpinLockMutex> guard(GetLock());
+    return GetSpanLimits()->AttributePerLinkCountLimit;
+  }
+
+private:
+  static nostd::shared_ptr<SpanLimits> &GetSpanLimits() noexcept
+  {
+    static nostd::shared_ptr<SpanLimits> spanLimits(new SpanLimits);
+    return spanLimits;
+  }
+
+  static opentelemetry::common::SpinLockMutex &GetLock() noexcept
+  {
+    static opentelemetry::common::SpinLockMutex lock;
+    return lock;
+  }
+
+  int AttributeCountLimit         = 128;
```

Comment:
Sorry for naïve question :

1. are we anticipating Span limits for different exporters (Jaeger, Zipkin, OTLP, or some other custom exporter, say FluentD-MsgPack) to be the same, and following these exact defaults?

2. If the answer to Q1 is `No`, then can we have these limits make somehow tunable, and be part of some form of Exporter configuration? This is becoming a bit tricky though. In case of multi-processor, in case if we want to route our telemetry to different exporter, and if every exporter has its own limits..

Apologies if it's a dumb question.

Reply:
Spec mentions these limits set to tracer provider and we have one global tracer provider, so in the current scenario all of these limits should be same. 
Moreover, according to my understanding since these limits are there for the safety from erroneous code adding absurd number of attributes to a span so it should be related to span related limits rather than depending upon any exporters. So, once these limits are defined they should be followed for each kind of span (sdk/exporters).